src/Serializer/RepresentationSerializerInterface.php

```diff
@@ -18,7 +18,7 @@ interface RepresentationSerializerInterface
      *
      * @param mixed $value
      *
-     * @return string|object|array|null
+     * @return array|string|object|bool|float|int|null
```

Comment:
This is wrong, see the PHPDoc above.

Reply:
I'm not sure if it was really wrong as it was reported by PHPStan itself. To make it happy without ignoring the error I added additional code to handle the possibility that those types are returned by the inner function